src/sparseml/transformers/finetune/data/custom.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+
+from datasets.dataset_dict import DatasetDict
+
+from sparseml.transformers.finetune.data import TextGenerationDataset
+
+
+@TextGenerationDataset.register(name="custom", alias=["json", "csv"])
```

Comment:
would this fail if we try to load `custom` instead of json or csv?

Reply:
Depends on the scope of the question. With respec to this particular code, no. 
Wrt the tests (loading the dataset), yes it will fail bc the first arg to hf's load_dataset must be json, csv, etc. Also then loading the dataset dict given a path, that will return an empty dict, bc no extension of custom matches with data.json, data.csv. ...